api/kv_v2.go

```diff
@@ -30,6 +30,26 @@ type KVMetadata struct {
 	Raw      *Secret
 }
 
+// KVMetadataInput is the subset of metadata that can be manually modified for
+// a KV v2 secret. This can also be passed to the PutMetadata method to create
+// a secret without any data yet.
+//
+// The struct's fields are all pointers. When used with the PatchMetadata
+// method, a pointer to a field's zero value (e.g. false for *bool) implies
+// that field should be reset to its zero value after update, whereas a field
+// set to a null pointer (e.g. null for *bool) implies the field should remain
+// unchanged.
+//
+// Since maps are already pointers, use an empty map to remove all
+// custom metadata. To explicitly set the zero value for time.Duration,
```

Comment:
Passing an empty map to patch custom metadata will actually leave the original value unchanged.

Reply:
Thanks for the thorough review, including of my explanations! This comment is actually in reference to how I'm asking the user to supply it in a KVMetadataInput struct, which (as I explain [here](https://github.com/hashicorp/vault/pull/15755/files#r888240307)), is not actually what gets sent to the /metadata endpoint. 

If you look at the toMetadataMap function, you'll see that I actually take this empty map and change it to a `nil` so that the /metadata endpoint behaves correctly, whereas I don't include nils from the KVMetadataInput struct at all:

```
	if metadata.CustomMetadata != nil {
		if len(metadata.CustomMetadata) == 0 { // empty non-nil map means delete all the keys
			md[customMetadataKey] = nil
		} else {
			md[customMetadataKey] = metadata.CustomMetadata
		}
	}
```

Since the call to the /metadata endpoint is behind the scenes, I figure making the KVMetadataInput struct as easy to use as possible regardless of how our backend endpoint is designed is most important (as in, letting users only set the fields they want to change, and leaving the others untouched i.e. nil.)

This is easy for users imo but hard for us devs and reviewers to understand at a glance, especially when familiar with how the /metadata endpoints work behind the scenes, so let me know if this still doesn't make sense!

Reply:
Oh gotcha, my mistake! Thanks for the clarification.